Compilation Issues

I used this a few days ago without any issue. Today when I try to verify the firmware code it fails. Below you'll find a copy/paste of the compilation attempt. 

voodoo.cpp:37:0: warning: "DEBUG" redefined [enabled by default]
 
 ^
In file included from ../services/inc/debug.h:1:0,
                 from ../wiring/inc/spark_wiring.h:36,
                 from ./inc/application.h:36,
                 from voodoo.cpp:35:
../services/inc/service_debug.h:109:0: note: this is the location of the previous definition
 #define DEBUG(fmt, ...)
 ^
voodoo.cpp: In function 'long unsigned int pingRead(int)':
voodoo.cpp:422:45: error: call of overloaded 'pulseIn(int&, PinState)' is ambiguous
 
                                             ^

voodoo.cpp:422:45: note: candidates are:
In file included from ./inc/application.h:36:0,
                 from voodoo.cpp:35:
../wiring/inc/spark_wiring.h:82:10: note: uint32_t pulseIn(pin_t, uint16_t)
 uint32_t pulseIn(pin_t pin, uint16_t value);
          ^
voodoo.cpp:375:15: note: long unsigned int pulseIn(uint16_t, uint8_t)
  * Original pulseIn Function for the Spark Core - Version 0.1.1 (Beta)
               ^
voodoo.cpp: In function 'int ToServoIndex(int)':
voodoo.cpp:242:1: warning: control reaches end of non-void function [-Wreturn-type]
     if (pin == 0 || pin == 1 || pin == 2 || pin == 3) return pin;
 ^
make[1]: *** [../build/target/user/platform-6voodoo.o] Error 1
make: *** [user] Error 2
Error: Could not compile. Please review your code.

--- a/firmware/voodoospark.cpp
+++ b/firmware/voodoospark.cpp
@@ -419,7 +419,7 @@ unsigned long pingRead(int pin) {
 
   pinMode(pin, INPUT);
 
-  unsigned long duration = pulseIn(pin, HIGH);
+  unsigned long duration = pulseIn((uint16_t)pin, (uint8_t)HIGH);
 
   return duration;
 }

